File: Algorithm_258_Add Digits.cpp
#include <iostream>
using namespace std;
int addDigits(int num)
{
	int tmp=num;
	int sum=num;
	while (tmp/10>0)
	{
		sum=0;
		while (tmp!=0)
		{
			sum+=tmp%10;
			tmp/=10;
		}
		tmp=sum;
	}
	return sum;
}
int main()
{
	cout<<addDigits(38);
	return 0;
}

File: Algorithm_258_Add Digits.java
// 方法：直接做

class Solution {
    public int addDigits(int num) {
        while (num / 10 > 0) {
            int tmp = num, newNum = 0;
            while (tmp > 0) {
                newNum += tmp % 10;
                tmp /= 10;
            }
            num = newNum;
        }
        return num;
    }
}